packages/chain-adapters/src/utxo/UtxoBaseAdapter.ts

```diff
@@ -42,67 +41,85 @@ import { bnOrZero } from '../utils/bignumber'
 import { GetAddressInput } from './types'
 import { utxoSelect } from './utxoSelect'
 
-export const UTXOChainIds = [KnownChainIds.BitcoinMainnet, KnownChainIds.DogecoinMainnet] as const
+export const utxoChainIds = [
+  KnownChainIds.BitcoinMainnet,
+  KnownChainIds.DogecoinMainnet,
+  KnownChainIds.LitecoinMainnet
+] as const
 
-export type UtxoChainId = typeof UTXOChainIds[number]
+export type UtxoChainId = typeof utxoChainIds[number]
 
-/**
- * Currently, we don't have a generic interact for UTXO providers, but will in the future.
- * Leaving this as-is for now, but we will need to test in the future when we have additional
- * UTXO chains implemented in unchained.
- */
 export interface ChainAdapterArgs {
+  chainId?: UtxoChainId
+  coinName: string
   providers: {
-    http: unchained.bitcoin.V1Api | unchained.dogecoin.V1Api
-    ws: unchained.ws.Client<unchained.bitcoin.BitcoinTx | unchained.dogecoin.BitcoinTx>
+    http: unchained.bitcoin.V1Api | unchained.dogecoin.V1Api | unchained.litecoin.V1Api
+    ws: unchained.ws.Client<unchained.utxo.types.Tx>
   }
-  coinName: string
-  chainId?: UtxoChainId
 }
 
-/**
- * Base chain adapter for all UTXO chains. When extending please add your ChainId to the
- * UTXOChainIds. For example:
- *
- * `export type UTXOChainIds = KnownChainIds.BitcoinMainnet | KnownChainIds.Litecoin`
- */
-export abstract class UTXOBaseAdapter<T extends UtxoChainId> implements IChainAdapter<T> {
-  protected chainId: UtxoChainId
-  protected assetId: AssetId
-  protected coinName: string
-  protected accountAddresses: Record<string, Array<string>> = {}
-  protected readonly supportedChainIds: ChainId[]
+export interface UtxoBaseAdapterArgs extends ChainAdapterArgs {
+  defaultBIP44Params: BIP44Params
+  defaultUtxoAccountType: UtxoAccountType
+  supportedChainIds: Array<ChainId>
+  supportedAccountTypes: Array<UtxoAccountType>
+  chainId: UtxoChainId
+}
+
+export abstract class UtxoBaseAdapter<T extends UtxoChainId> implements IChainAdapter<T> {
+  protected readonly chainId: UtxoChainId
+  protected readonly coinName: string
```

Comment:
don't address - wtf is this even for any mroe?

Reply:
hdwallet goodness